tests/foreman/cli/test_auth.py

```diff
@@ -0,0 +1,417 @@
+        self.assertIn(
+            LOGEDIN_MSG.format(self.uname_admin),
+            result[0][u'message']
+        )
+        # list organizations without supplying credentials
+        with self.assertNotRaises(CLIReturnCodeError):
+            Org.list(pass_credentials=False)
+        Auth.login({'username': self.uname_viewer, 'password': self.password})
+        result = Auth.status()
+        self.assertIn(
+            LOGEDIN_MSG.format(self.uname_viewer),
+            result[0][u'message']
+        )
+        with self.assertNotRaises(CLIReturnCodeError):
+            Org.list(pass_credentials=False)
+
+    @tier1
+    def test_positive_session_survives_unauthenticated_call(self):
+        """Check if session stays up after unauthenticated call
+
+        :id: 8bc304a0-70ea-489c-9c3f-ea8343c5284c
+
+        :Steps:
+
+            1. Set use_sessions
+            2. Authenticate, assert credentials are not demanded
+               on next command run
+            3. Run `hammer ping`
+
+        :expectedresults: The session is unchanged
+
+        """
+        result = configure_sessions()
+        self.assertEqual(result, 0, 'Failed to configure hammer sessions')
+        Auth.login({'username': self.uname_admin, 'password': self.password})
+        result = Auth.status()
+        self.assertIn(
+            LOGEDIN_MSG.format(self.uname_admin),
+            result[0][u'message']
+        )
+        # list organizations without supplying credentials
+        with self.assertRaises(CLIReturnCodeError):
+            Org.list(pass_credentials=False)
+        ssh.command('hammer ping')
+        result = Auth.status()
+        self.assertIn(
+            LOGEDIN_MSG.format(self.uname_admin),
+            result[0][u'message']
+        )
+        with self.assertRaises(CLIReturnCodeError):
+            Org.list(pass_credentials=False)
+
+    @tier1
+    @skip_if_bug_open("bugzilla", "1465552")
+    def test_positive_session_survives_failed_login(self):
+        """Check if session stays up after failed login attempt
+
+        :id: 6c4d5c4c-eff0-411b-829f-0c2f2ec26132
+
+        :BZ: 1465552
+
+        :Steps:
+
+            1. Set use_sessions
+            2. Authenticate, assert credentials are not demanded
+               on next command run
+            3. Run login with invalid credentials
+
+        :expectedresults: The session is unchanged
+
+        """
+        result = configure_sessions()
+        self.assertEqual(result, 0, 'Failed to configure hammer sessions')
+        Auth.login({'username': self.uname_admin, 'password': self.password})
+        result = Auth.status()
+        self.assertIn(
+            LOGEDIN_MSG.format(self.uname_admin),
+            result[0][u'message']
+        )
+        with self.assertRaises(CLIReturnCodeError):
+            Org.list(pass_credentials=False)
+        try:
+            Auth.login({
+                'username': self.uname_viewer,
+                'password': gen_string('alpha')})
```

Comment:
I think you should remove the `try`-`except` block and wrap the login command with `assertNotRaises`. Otherwise the test will pass silently if login succeeds.